contracts/DVA/DVATransferManager.sol

```diff
@@ -0,0 +1,651 @@
+
+    error RecipientIsNotVerified(address _tokenAddress, address _recipient);
+
+    error DVAManagerIsNotVerifiedForTheToken(address _tokenAddress);
+
+    error TokenTransferFailed(address _tokenAddress, address _from, address _to, uint256 _amount);
+
+    error InvalidTransferID(bytes32 _transferID);
+
+    error TransferIsNotInPendingStatus(bytes32 _transferID);
+
+    error ApprovalsMustBeSequential(bytes32 _transferID);
+
+    error ApproverNotFound(bytes32 _transferID, address _approver);
+
+    error SignaturesCanNotBeEmpty(bytes32 _transferID);
+
+    constructor(){
+        _txNonce = 0;
+    }
+
+    /**
+    *  @dev modify the approval criteria of a token
+     *  @param tokenAddress is the token address.
+     *  @param includeRecipientApprover determines whether the recipient is included in the approver list
+     *  @param includeAgentApprover determines whether the agent is included in the approver list
+     *  @param sequentialApproval determines whether approvals must be sequential
+     *  @param additionalApprovers are the addresses of additional approvers to be added to the approver list
+     *  Only an agent of a token can call this function
+     *  DVATransferManager must be an agent of the given token
+     *  emits an `ApprovalCriteriaModified` event
+     */
+    function modifyApprovalCriteria(
+        address tokenAddress,
+        bool includeRecipientApprover,
+        bool includeAgentApprover,
+        bool sequentialApproval,
+        address[] memory additionalApprovers
+    ) external {
+        if (!AgentRole(tokenAddress).isAgent(msg.sender)) {
+            revert OnlyTokenAgentCanCall(tokenAddress);
+        }
+
+        if (!IToken(tokenAddress).identityRegistry().isVerified(address(this))) {
+            revert DVAManagerIsNotVerifiedForTheToken(tokenAddress);
+        }
+
+        bytes32 hash = keccak256(
+            abi.encode(
+                tokenAddress,
+                includeRecipientApprover,
+                includeAgentApprover,
+                additionalApprovers
+            )
+        );
+
+        _approvalCriteria[tokenAddress] = ApprovalCriteria(
+            tokenAddress,
+            includeRecipientApprover,
+            includeAgentApprover,
+            sequentialApproval,
+            additionalApprovers,
+            hash);
+
+        emit ApprovalCriteriaModified(
+            tokenAddress,
+            includeRecipientApprover,
+            includeAgentApprover,
+            sequentialApproval,
+            additionalApprovers,
+            hash
+        );
+    }
+
+    /**
+     *  @dev initiates a new transfer
+     *  @param tokenAddress is the address of the token
+     *  @param recipient is the address of the recipient
+     *  @param amount is the transfer amount
+     *  Approval criteria must be preset for the given token address
+     *  Receiver must be verified for the given token address
+     *  emits a `TransferInitiated` event
```

Comment:
could add that the sender should give allowance first to the DvA smart contract address on the token contract, and that this allowance needs to be at least `amount`